enterprise/app/services/enterprise/auto_assignment/assignment_service.rb

```diff
@@ -0,0 +1,101 @@
+module Enterprise::AutoAssignment::AssignmentService
+  private
+
+  # Override assignment config to use policy if available
+  def assignment_config
+    return super unless policy
+
+    {
+      'conversation_priority' => policy.conversation_priority,
+      'fair_distribution_limit' => policy.fair_distribution_limit,
+      'fair_distribution_window' => policy.fair_distribution_window,
+      'balanced' => policy.balanced?
+    }.compact
+  end
+
+  # Override to check policy first
+  def assignment_enabled?
+    return policy.enabled? if policy
+
+    super
+  end
+
+  # Extend agent finding to add capacity checks
+  def find_available_agent
+    agents = filter_agents_by_rate_limit(inbox.available_agents)
+    agents = filter_agents_by_capacity(agents) if capacity_filtering_enabled?
```

Comment:
I have a feeling this will be a perf bottleneck in the future, since the capacity check does a DB count for each conversation. 

While it will make it complex, we will need a precompute + update mechanism like we do it for rate limiting.  

@sojan-official do you think this will an issue?